File: Client/src/app/content/notes/models/fullNote.ts
import { NoteTypeENUM } from 'src/app/shared/enums/NoteTypesEnum';
import { RefTypeENUM } from 'src/app/shared/enums/refTypeEnum';
import { Label } from '../../labels/models/label';

export interface FullNote {
  id: string;
  title: string;
  color: string;
  labels: Label[];
  refTypeId: RefTypeENUM;
  noteTypeId: NoteTypeENUM;
  isLocked: boolean;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date;
}

File: Backend/WriteAPI/Common/DTO/notes/FullNote.cs
using Common.DatabaseModels.models.Notes;
using Common.DatabaseModels.models.Systems;
using Common.DTO.app;
using Common.DTO.labels;
using System;
using System.Collections.Generic;

namespace Common.DTO.notes
{
    public class FullNote
    {
        public Guid Id { get; set; }
        public string Title { set; get; }
        public string Color { set; get; }
        public List<LabelDTO> Labels { set; get; }
        public RefTypeENUM RefTypeId { set; get; }
        public NoteTypeENUM NoteTypeId { set; get; }
        public bool IsLocked { set; get; }

        public DateTimeOffset DeletedAt { set; get; }
        public DateTimeOffset UpdatedAt { set; get; }
        public DateTimeOffset CreatedAt { set; get; }
    }
}
